components/camel-jms/src/main/java/org/apache/camel/component/jms/reply/ReplyManagerSupport.java

```diff
@@ -267,7 +269,8 @@ protected void doStart() throws Exception {
         // timeout map to use for purging messages which have timed out, while waiting for an expected reply
         // when doing request/reply over JMS
         log.trace("Using timeout checker interval with {} millis", endpoint.getRequestTimeoutCheckerInterval());
-        correlation = new CorrelationTimeoutMap(scheduledExecutorService, endpoint.getRequestTimeoutCheckerInterval(), executorService);
+        TimeoutMap<String, ReplyHandler> timeoutMap = new DefaultTimeoutMap<>(scheduledExecutorService, endpoint.getRequestTimeoutCheckerInterval());
```

Comment:
Its a bit map in map in map ;). Just wonder if we could avoid all the delegate/wrapping, and let it be so you can extend DefaultTimeoutMap as before

Reply:
Committed the change requested. I looked at both extensions and their additional function:
1 trace logging - can be implemented in the local listener
2 putIfAbsent() logging is redundant as the (sole) calling method repeats it org.apache.camel.component.rabbitmq.reply.ReplyManagerSupport#registerReply
3 Support for -1 maximum timeout. This could be generalised into `DefaultTimeoutMap`? Currently default supports all negative timeouts as meaning "accept but evict immediately in next poll". There is a change in semantics if we mean negative to say "never evict" (+need to decide if this means `Integer.MAX_VALUE` or `Long.MAX_VALUE`)

I have implemented 1 and 2. If we go for 3 then both extensions become nothing more than Factory Methods constructing a `DefaultTimeoutMap`